cookbook/persistance/key-value.md

```diff
@@ -0,0 +1,169 @@
+---
+layout: page
+title: "Storing key-value data on disk"
+permalink: /cookbook/persistence/key-value/
+---
+
+If we have a relatively small collection of key-values that we'd like to save, we can use the [shared_preferences](https://pub.dartlang.org/packages/shared_preferences) plugin.
+
+Normally we would have to write native platform integration for storing data for both platforms. Fortunately the [shared_preferences](https://pub.dartlang.org/packages/shared_preferences) plugin can be used for that. The shared preferences plugin wraps `NSUserDefaults` on iOS and `SharedPreferences` on Android, providing a persistent store for simple data.
+
+## Setup
+
+Before we start, we need to add the [shared_preferences](https://pub.dartlang.org/packages/shared_preferences) plugin to our `pubspec.yaml` file:
+
+```yaml
+dependencies:
+  flutter:
+    sdk: flutter
+  shared_preferences: "<newest version>"
+```
+
+## Saving data
+
+To persist key-value data, we can use `SharedPreferences` class.
+For saving data we call `set` method. Note that data is persisted asynchronously. If we want to get notified when data is saved use `commit()` function. 
+
+```dart
+// obtain shared preferences 
+SharedPreferences prefs = await SharedPreferences.getInstance();
+
+// set new value
+prefs.setInt('counter', counter);
+```
+
+
+## Reading data
+
+```dart
+SharedPreferences prefs = await SharedPreferences.getInstance();
+
+int counter = (prefs.getInt('counter') ?? 0) + 1;
+```
+
+In example above we load data from the `counter` key, if it not exists, `0` is returned. 
```

Comment:
"In this example we'll load data from the counter key. If it *does not* exist, 0 is returned."